### PR TITLE
feat!: bump to Python3.9

### PR DESCRIPTION
**Pull request by sebastianliebscher**
### SUMMARY
Running CI integration and unit tests for Python 3.9 is a first step to fully upgrade official release to Python 3.10. 
This PR bumps Python to version 3.9:

- [docker_build_push.sh](https://github.com/apache/superset/pull/23890/files#diff-e1276b2a2c85f7fb3b628b7acadbc417edb788f2848624f5f8180837bb3f1c53) now uses the PY_VER ( 3.9.**16** ) from the Dockerfile instead of version 3.8 to build reproducible build images
- GitHub workflows now using **Python 3.9** instead of 3.8:
  - [asf.yaml](https://github.com/apache/superset/pull/23890/files#diff-b4c2a69650f9ac84008ad9f745859a645c9466350ffdfe5f919655039ee2e2c3) (required checks)
  - [superset-applitool-cypress.yml](https://github.com/apache/superset/pull/23890/files#diff-659c7e0d194ca66c1aab66027a1a742b104003821640707cab76c4947c310e1f)
  - [superset-cli.yml](https://github.com/apache/superset/pull/23890/files#diff-2b75e0452e0536fda5beaf39306be5ed9df9f7f6c30b1594231642b4d0b8668a)
  - [superset-e2e.yml](https://github.com/apache/superset/pull/23890/files#diff-4fb9e27b16994afdc00157730374031d48422742b39af76e079c12c0e14b1cd4)
  - [superset-helm-lint.yml](https://github.com/apache/superset/pull/23890/files#diff-5584a0028fd754716fec92d0315ee36bfd3cbc5c989d24b03508cfe7a8fafab5)
  - [superset-python-integrationtest.yml](https://github.com/apache/superset/pull/23890/files#diff-f40cc40023f61daa947bf50d3e8b3807e19ee0eb5204fcc5e82732c0933f3b0b) (integration tests)
  - [superset-python-misc.yml](https://github.com/apache/superset/pull/23890/files#diff-09d91a25ccd41706d36bad437e4e3d9d93c909135cb26e4996aaa7e543deea4b)
  - [superset-python-presto-hive.yml](https://github.com/apache/superset/pull/23890/files#diff-c341eabb1f56a2e06d6b48f1ba56c3863b893e3da4f539ff4cfe6af10a0a3c1b)
  - [superset-python-unittest.yml](https://github.com/apache/superset/pull/23890/files#diff-11c8ef05ec3f4248016ecbb54735721cd07a298e97ad7c73eb028d38c51d4949) (unit tests)
  - [superset-translations.yml](https://github.com/apache/superset/pull/23890/files#diff-a32ee96884fb403fb3442d499a1225ead0bedd8593f3f45435c9a680d171b776)
- GitHub workflows running additionally for Python 3.10:
  - [superset-python-integrationtest.yml](https://github.com/apache/superset/pull/23890/files#diff-f40cc40023f61daa947bf50d3e8b3807e19ee0eb5204fcc5e82732c0933f3b0b)
   - [superset-python-unittest.yml](https://github.com/apache/superset/pull/23890/files#diff-11c8ef05ec3f4248016ecbb54735721cd07a298e97ad7c73eb028d38c51d4949)
  

Also, added double quotes around `strategy.matrix.python-versions` values as without quotes, e.g. `3.10` is interpreted as `3.1` [reference to GitHub Actions Doc](https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#using-the-python-starter-workflow)

### TESTING INSTRUCTIONS
All existing and new CI jobs run without errors.

### CODE REVIEW

**Comment by villebro** [comment]
@sebastianliebscher is adding quotes around the python version the "correct" way? It would be nice to add a note about that in the description.

**Comment by sebastianliebscher** [comment]
> @sebastianliebscher is adding quotes around the python version the "correct" way? It would be nice to add a note about that in the description.

@villebro Without the quotes, e.g. `3.10` is interpreted as `3.1`. But you are right, I missed that in the desciption. I updated the desciption accordingly :+1:

**Comment by villebro** [comment]
> @villebro Without the quotes, e.g. `3.10` is interpreted as `3.1`. But you are right, I missed that in the desciption. I updated the desciption accordingly 👍

Thanks @sebastianliebscher - makes sense 👍

**Comment by EugeneTorap** [comment]
Hi @rusackas! Can you help us with the required CI tests for old Py3.8?

File: .asf.yaml [hunk 1]
```diff
@@ -66,12 +66,12 @@ github:
...
-          - pre-commit (3.8)
-          - python-lint (3.8)
-          - test-mysql (3.8)
-          - test-postgres (3.8)
+          - pre-commit (3.9)
+          - python-lint (3.9)
+          - test-mysql (3.9)
           - test-postgres (3.9)
-          - test-sqlite (3.8)
+          - test-postgres (3.9)
```
Comment (villebro) [inline .asf.yaml:73]: Shouldn't this be
```suggestion
          - test-postgres (3.10)
```

File: scripts/benchmark_migration.py [hunk 2]
```diff
@@ -120,7 +120,7 @@ def find_models(module: ModuleType) -> List[Type[Model]]:
     # sort topologically so we can create entities in order and
     # maintain relationships (eg, create a database before creating
     # a slice)
-    sorter = TopologicalSorter()
+    sorter = TopologicalSorter()  # type: TopologicalSorter[Any]
```
Comment (villebro) [inline scripts/benchmark_migration.py:123]: Wouldn't this be the preferred way of specifying the type?
```python
    sorter: TopologicalSorter[Any] = TopologicalSorter() 
```

**Review by villebro — COMMENTED** [review]
A few quick comments

File: .asf.yaml (hunk 1 above)
Comment (EugeneTorap) [reply .asf.yaml:73]: @villebro I have done it in #24002

File: Dockerfile [hunk 3]
```diff
@@ -18,7 +18,7 @@
 ######################################################################
 # Node stage to deal with static asset construction
 ######################################################################
-ARG PY_VER=3.8.16-slim
+ARG PY_VER=3.9.16-slim
```
Comment (dpgaspar) [inline Dockerfile:21]: checked that 3.9.16 is the latest 3.9

**Review by dpgaspar — APPROVED** [review]
LGTM
thank you @sebastianliebscher this is great!

**Review by villebro — APPROVED** [review]
LGTM, exciting! 🚀 @sebastianliebscher let's rebase to restart CI

**Comment by eschutho** [comment]
I marked this as a breaking change. All good for now since we can get it into the 3.0 release.

File: UPDATING.md [hunk 4]
```diff
@@ -46,6 +46,7 @@ assists people when migrating to a new version.
 - [21163](https://github.com/apache/superset/pull/21163): The time grain will be decoupled from the time filter column and the time grain control will move below the X-Axis control when `GENERIC_CHART_AXES` feature flags set to `True`. The time grain will be applied on the time column in the column-like controls(x axis, dimensions) instead of the time column in the time section.
 - [21284](https://github.com/apache/superset/pull/21284): The non-functional `MAX_TABLE_NAMES` config key has been removed.
 - [21794](https://github.com/apache/superset/pull/21794): Deprecates the undocumented `PRESTO_SPLIT_VIEWS_FROM_TABLES` feature flag. Now for Presto, like other engines, only physical tables are treated as tables.
+- [23890](https://github.com/apache/superset/pull/23890): Removes Python 3.8 support.
```
Comment (eschutho) [inline UPDATING.md:49]: I need to create a PR to update this file for 2.1, so I'll move this into Breaking Changes then.

**Pull request merged**